src/yunohost/io.py

```diff
@@ -0,0 +1,222 @@
+    along with this program; if not, see http://www.gnu.org/licenses
+
+    yunohost_certificate.py
+
+    Manage certificates, in particular Let's encrypt
+"""
+
+import os
+import errno
+import pwd
+import grp
+import requests
+import subprocess
+import socket
+import glob
+
+import json
+
+from moulinette.core import MoulinetteError
+from moulinette.utils.log import getActionLogger
+
+logger = getActionLogger('yunohost.io')
+
+
+def _read_file(file_path):
+    """
+    Read a regular text file
+
+    Keyword argument:
+        file_path -- Path to the json file
+    """
+    assert isinstance(file_path, str)
+
+    # Check file exist
+    if not os.path.isfile(file_path):
+        raise MoulinetteError(errno.EROENT,
+                              m18n.n('io_no_such_file', file=file_path))
+
+    # Open file and read content
+    try:
+        with open(file_path, "r") as f:
+            file_content = f.read()
+    except IOError as e:
+        raise MoulinetteError(errno.EACCES,
+                              m18n.n('io_cannot_open_file', file=file_path))
+    except Exception as e:
+        raise MoulinetteError(errno.EIO,
+                              m18n.n('io_unknown_error_opening_file',
+                                     file=file_path, error=str(e)))
+
+    return file_content
+
+
+def _read_json(file_path):
+    """
+    Read a json file
+
+    Keyword argument:
+        file_path -- Path to the json file
+    """
+    assert isinstance(file_path, str)
+
+    # Read file
+    file_content = _read_file(file_path)
+
+    # Try to load json to check if it's syntaxically correct
+    try:
+        loaded_json = json.loads(file_content)
+    except ValueError:
+        raise MoulinetteError(errno.EINVAL,
+                              m18n.n('io_corrupted_json', file=file_path))
+
+    return loaded_json
+
+
+def _write_json(file_path, input_dict):
+    """
+    Write a dictionnary to a json file
+
+    Keyword argument:
+        file_path -- Path to the json file
+        input_dict -- The dictionnary to write
+    """
+
+    # Assumptions
+    assert isinstance(file_path, str)
+    assert isinstance(input_dict, dict)
+
+    # Write dict to file
+    try:
+        with open(file_path, "w") as f:
+            json.dump(input_dict, f)
+    except IOError as e:
+        raise MoulinetteError(errno.EACCES,
+                              m18n.n('io_cannot_write_file', file=file_path))
+    except Exception as e:
+        raise MoulinetteError(errno.EIO,
+                              m18n.n('io_unknown_error_writing_file',
+                                     file=file_path, error=str(e)))
+
+
+def _remove_file(file_path):
+    """
+    Remove a regular file if it exists
+
+    Keyword argument:
+        file_path -- Path of the file to remove
+    """
+    # Assumptions
+    assert isinstance(file_path, str)
+
+    if os.path.exists(file_path):
+        try:
+            os.remove(file_path)
```

Comment:
`os.remove` will probably raise an IOError or OSError exception that could be catched.